Client/Frontend/Browser/ErrorPageHelper.swift

```diff
@@ -187,15 +192,12 @@ class ErrorPageHelper {
         return false
     }
 
-    class func decodeURL(url: NSURL) -> NSURL? {
-        let query = url.getQuery()
-        if let queryUrl = query["url"] {
-            let escaped = NSURL(string: queryUrl.unescape())
-            if let escaped = escaped where isErrorPageURL(escaped) {
-                return decodeURL(escaped)
-            }
-            return escaped
+    class func originalURLFromQuery(url: NSURL) -> NSURL? {
+        let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false)
+        if let queryURL = components?.queryItems?.filter({ $0.name == "url" }).first?.value {
```

Comment:
Nit: We have a `find` utility function on Arrays now that does this https://github.com/mozilla/firefox-ios/blob/master/Utils/Extensions/ArrayExtensions.swift#L9